File: WebApplication1/wwwroot/js/Calidad.js
$(document).ready(function () {
    dataTable = $("#tblCalidad").DataTable({
        "ajax": {
            "url": "/Calidades/GetTodosCalidad",
            "type": "GET",
            "dataType": "json",
        },
        "columns": [
            { "data": "idCalidad", "width": "10%" },
            { "data": "modelo", "width": "20%" },
            { "data": "fechaDeRecepcion", "width": "30%", "render": function (data) { return moment(data).format('DD/MM/YYYY') } },
            { "data": "fechaRevision", "width": "30%", "render": function (data) { return moment(data).format('DD/MM/YYYY') } },
            { "data": "estado", "width": "30%" },
            {
                "data": "fechaEnvioMaquila", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            {
                "data": "fechaRecepcionRechazo", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            { "data": "incidencia", "width": "30%" },
            { "data": "area", "width": "30%" },
            {
                "data": "idCalidad",
                "render": function (data, type, row) {
                    let btnClass = row.fechaRecepcionRechazo ? "btn-secondary" : "btn-success";
                    let btnText = row.fechaRecepcionRechazo ? "Editado" : "Editar";
                    console.log("Fecha de Recepcion Rechazo: ", row.fechaRecepcionRechazo);  // Verifica el valor
                    return `<div class="text-center">
                        <a href="/Calidades/Edit/${data}" class="btn ${btnClass} text-white" style="cursor: pointer;">
                            <i class="fas fa-edit"></i> ${btnText}
                        </a>
                    </div>`;
                },
                "width": "20%"
            }
        ],
        "language": {
            "search": "Buscar:",
        },
        "dom": '<"top"f>rt<"bottom"ip><"clear">',
        "searching": true,
        "scrollX": true,
        "paging": false,
        "info": false,
    });
    $('#tblCalidad_filter').detach().appendTo('.d-flex');
});

File: WebApplication1/Models/Calidad.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebTabitas.Models
{
    public class Calidad
    {
        public int IdCalidad { get; set; }
        public int IdGeneral { get; set; }
        public string Modelo { get; set; }
        [Required(ErrorMessage = "La Fecha de Recepcion es obligatorio")]
        public DateTime FechaDeRecepcion { get; set; } = DateTime.Now;
        [Required(ErrorMessage = "La Fecha de Recepcion es obligatorio")]
        public DateTime FechaRevision { get; set; } = DateTime.Now;
        [Required(ErrorMessage = "El Estado es obligatorio")]
        public string Estado { get; set; }
        public DateTime? FechaEnvioMaquila { get; set; }
        public DateTime? FechaRecepcionRechazo { get; set; }
        public string? Incidencia { get; set; }
        public int IdProceso { get; set; }
        public string Area { get; set; }
    }
}
